### PR TITLE
Move int-set to read_fonts::collections::intset

### PR DESCRIPTION
**Pull request by cmyr**
This is gated on `std`. Chad keeps mentioning that we really only need alloc, but that's trickier, and we're already using `std` as a feature for closure in GSUB.

This also requires some changes to make the colr/cmap closure code compile; all that is now also gated on `std`.

### CODE REVIEW

File: read-fonts/src/collections.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//! data structures useful for font work
```
Comment (dfrg) [inline read-fonts/src/collections.rs:1]:
```suggestion
//! Data structures useful for font work.
```
Hurts my eyes when public docs are missing capitalization and punctuation :)

File: read-fonts/src/collections.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+pub mod intset;
+pub use intset::IntSet;
```
Comment (dfrg) [inline read-fonts/src/collections.rs:4]: `std` uses `hash_set` and `btree_set` so I think we should follow with `int_set` just for consistency :shrug:

**Review by dfrg — APPROVED** [review]
Thanks for getting this done!

Per IM, we may want to take on an unconditional dependency on `alloc` and just provide this functionality without a feature gate but I think this is perfectly fine for now.

Left the two tiniest nits below.

File: read-fonts/src/collections.rs (hunk 2 above)
Comment (cmyr) [reply read-fonts/src/collections.rs:4]: oh I went with `intset` _specifically_ because I thought that's what `std` does. 🙄

**Pull request merged**